internal/dirs/copy.go

```diff
@@ -18,5 +22,49 @@ func OverlayDir(dir string) error {
 
 	slog.Info("Copying Overlay directory's files", "dir", dir)
 
-	return cp.Copy(Overlay, dir)
+	return cp.Copy(overlay, dir, cp.Options{
+		Sync: true,
+		PreserveOwner: true,
+		PreserveTimes: true,
+		Skip: func(info os.FileInfo, src, dest string) (bool, error) {
+			if info.IsDir() {
+				return false, nil
+			}
+
+			if info.Mode()&os.ModeSymlink != 0 {
+				destPath, err := filepath.EvalSymlinks(dest)
+				if err != nil {
+					return false, err
+				}
+
+				srcPath, err := filepath.EvalSymlinks(src)
+				if err != nil {
+					return false, err
+				}
+
+				if destPath == srcPath {
+					return true, nil
+				}
```

Comment:
turn this into a general **internal** function that can be used both here and in tests.